internal/lagoon/compose.go

```diff
@@ -57,6 +59,9 @@ func UnmarshalLagoonDockerComposeYAML(file string) ([]OriginalServiceOrder, erro
 		// extract the services only
 		if item.Key.(string) == "services" {
 			for idx, v := range item.Value.(goyaml.MapSlice) {
+				if err := utilvalidation.IsDNS1035Label(v.Key.(string)); err != nil {
```

Comment:
I think we should also then check if the `lagoon.type` label is not `none` here too. Because `lagoon.type=none` do not get deployed, we shouldn't fail builds for people using services like `wait_dependencies` for containers they may use for local development only that don't make it into lagoon.